client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/test/index.tsx

```diff
@@ -78,4 +76,48 @@ describe( 'SiteMigrationInstructions', () => {
 
 		expect( deleteMigrationSticker ).toHaveBeenCalledWith( 123 );
 	} );
+
+	it( 'should render the steps progress', () => {
+		const { container } = render();
+
+		expect( container.querySelector( '.circular__progress-bar-text' )?.innerHTML ).toEqual( '0/3' );
+	} );
+
+	it( 'should update the progress when completing steps', async () => {
+		const { container, getByRole } = render();
+
+		await userEvent.click( getByRole( 'button', { name: /Next/ } ) );
+
+		expect( container.querySelector( '.circular__progress-bar-text' )?.innerHTML ).toEqual( '1/3' );
+	} );
+
+	it( 'should navigate to the next step when clicking on Next', async () => {
+		const { queryByText, getByRole } = render();
+
+		await userEvent.click( getByRole( 'button', { name: /Next/ } ) );
+
+		expect(
+			queryByText( 'Then, pick WordPress.com as your destination host.' )
+		).toBeInTheDocument();
+	} );
+
+	it( 'should be able to navigate back to the first step when it was completed', async () => {
+		const { queryByText, getByRole } = render();
+
+		await userEvent.click( getByRole( 'button', { name: /Next/ } ) );
+		await userEvent.click( getByRole( 'button', { name: /Install the Migrate Guru plugin/ } ) );
+
+		expect( queryByText( 'Migrate Guru plugin' ) ).toBeInTheDocument();
+	} );
+
+	it( 'should navigate to the next step when the steps are completed', async () => {
```

Comment:
This text will need some extra work when we work on [this issue](https://github.com/Automattic/wp-calypso/issues/91780). It will depend on some extra actions to be able to complete.